src/gradletestrunnersmoketest.sh

```diff
@@ -18,14 +18,30 @@ gradletestrunnersmoketest () {
   ./gradlew spotlessCheck
   ./gradlew spotbugsMain
 
+  outputDir="tmp/TestRunnerResults"
+
   echo "Running test suite"
-  ./gradlew runTest --args="suites/PRSmokeTests.json tmp/TestRunnerResults"
+  runTestOutput=$(./gradlew runTest --args="suites/PRSmokeTests.json $outputDir" 2>&1)
```

Comment:
Aside from @smark88 s approach, another thought I had (I'm not sure it's more readable but avoids output redirection and getting the error code using `$?`) is to do something like:

```suggestion
  ./gradlew runTest --args="suites/PRSmokeTests.json $outputDir" ||
    echo "Running test suite failed, Uploading results" &&
    ./gradlew uploadResults --args="BroadJadeDev.json $outputDir" && 
    exit 1
```

Reply:
Yeah, I had that originally but thought maybe the logic about checking the error code was clearer this way. Now I'm having second thoughts. I'll change it and see what you think!